rclcpp/include/rclcpp/parameter_client.hpp

```diff
@@ -134,6 +134,34 @@ class AsyncParametersClient
       std::make_shared<Alloc>());
   }
 
+  template<
+    typename CallbackT,
+    typename Alloc = std::allocator<void>,
+    typename SubscriptionT =
+    rclcpp::Subscription<rcl_interfaces::msg::ParameterEvent, Alloc>>
+  static typename rclcpp::Subscription<rcl_interfaces::msg::ParameterEvent>::SharedPtr
+  on_parameter_event(
+    CallbackT && callback,
+    rclcpp::node_interfaces::NodeTopicsInterface * node_topics)
+  {
+    using rclcpp::message_memory_strategy::MessageMemoryStrategy;
+    auto msg_mem_strat =
+      MessageMemoryStrategy<rcl_interfaces::msg::ParameterEvent, Alloc>::create_default();
+
+    using rcl_interfaces::msg::ParameterEvent;
+    return rclcpp::create_subscription<
```

Comment:
This is going to need to be updated to use `SubscriptionOptions`

Reply:
Updated to the new subscritpions API.
Sorry for the several force-push, but I had some issues with DCO